mysys/crc32/crc32_arm64.c

```diff
@@ -131,109 +139,122 @@ uint32_t crc32c_aarch64(uint32_t crc, const unsigned char *buffer, uint64_t len)
 
 	if (buffer) {
```

Comment:
Isn’t this condition redundant? When would this function be invoked with `buffer=NULL`?